src/reducer.js

```diff
@@ -0,0 +1,41 @@
+export const initialState = {
+  id: 100,
+  restaurants: [],
+  restautrant: {
+    name: '',
+    type: '',
+    address: '',
+  },
+};
+
+// TODO: updateRestautrantInfo 리듀서 기능 정의
+export default function reducer(state = initialState, action = {}) {
+  if (action.type === 'updateRestautrantInfo') {
+    const { key, value } = action.payload;
+
+    return {
+      ...state,
+      restautrant: {
+        ...state.restautrant,
+        [key]: value,
+      },
+    };
+  }
+
+  if (action.type === 'addRestaurant') {
+    const { restautrant } = state;
+    const { name, type, address } = restautrant;
+
+    if (!(name && type && address)) {
```

Comment:
```suggestion
    if (!name || !type || !address) {
```

은 어떨까요? 하나라도 없다면 종료한다는 것이 더 이해하기 쉬운 것 같아요